src/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useRef } from "react";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { uiColors } from "@leafygreen-ui/palette";
+import { Body } from "@leafygreen-ui/typography";
+import Icon from "components/Icon";
+import { useOnClickOutside } from "hooks";
+
+const { gray, white } = uiColors;
+
+interface DropdownProps {
+  ["data-cy"]?: string;
+  disabled?: boolean;
+  buttonRenderer?: () => React.ReactNode;
+  buttonText?: string;
+  children?: React.ReactNode;
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+}
+const Dropdown: React.FC<DropdownProps> = ({
+  "data-cy": dataCy = "dropdown-button",
+  disabled = false,
+  buttonText,
+  buttonRenderer,
+  children,
+  isOpen,
+  setIsOpen,
+}) => {
+  const listMenuRef = useRef(null);
+  const menuButtonRef = useRef(null);
+
+  // Handle onClickOutside
+  useOnClickOutside([listMenuRef, menuButtonRef], () => setIsOpen(false));
+
+  return (
+    <Container>
+      <StyledButton
+        ref={menuButtonRef}
+        onClick={() => setIsOpen(!isOpen)}
+        data-cy={dataCy}
+        id="searchable-dropdown"
+        disabled={disabled}
+        rightGlyph={<Icon glyph={isOpen ? "ChevronUp" : "ChevronDown"} />}
+      >
+        <ButtonContent>
+          <LabelWrapper>
+            {buttonRenderer ? (
+              buttonRenderer()
+            ) : (
+              <OverflowBody data-cy="dropdown-value">{buttonText}</OverflowBody>
+            )}
+          </LabelWrapper>
+        </ButtonContent>
+      </StyledButton>
+      {isOpen && (
+        <RelativeWrapper>
+          <OptionsWrapper ref={listMenuRef} data-cy={`${dataCy}-options`}>
+            {children}
+          </OptionsWrapper>
+        </RelativeWrapper>
+      )}
+    </Container>
+  );
+};
+
+interface DropdownWithRefProps
+  extends Omit<DropdownProps, "isOpen" | "setIsOpen"> {
+  ref?: React.Ref<DropdownWithRef>;
+}
+
+interface DropdownWithRefState {
+  isOpen: boolean;
+}
+/** DropdownWithRef is a class component that allows the implementer
+ *  to control its internal state methods with a ref in order to trigger state updates */
+class DropdownWithRef extends React.Component<
```

Comment:
Just to clarify, why exactly must this component be a class?

Reply:
I wanted a way for a  component implementing this to be able to close the dropdown. Without giving up the state control this component has. In order to do that i wanted to expose a method an implementer component could use to trigger a state change. 
To do this i needed to use a ref to expose the components internal methods. Functional components don't support this functionality.